packages/phoenix-ng/projects/phoenix-ui-components/lib/components/ui-menu/view-options/cartesian-grid-config/cartesian-grid-config.component.ts

```diff
@@ -32,12 +37,57 @@ export class CartesianGridConfigComponent implements OnInit {
     this.showCartesianGrid = this.data.gridVisible;
     this.scale = this.data.scale;
     this.gridConfig = this.eventDisplay.getUIManager().getCartesianGridConfig();
+    this.cartesianPos = this.eventDisplay.getThreeManager().origin;
   }
 
   onClose() {
     this.dialogRef.close();
   }
 
+  onSave(x, y, z) {
+    this.shiftCartesianGridByValues(new Vector3(x * 10, y * 10, z * 10));
+  }
+
+  shiftCartesianGridByPointer(change: MatCheckboxChange) {
```

Comment:
Actually, I made the grid translatable, by two methods, one, by value, another, by pointer.
By value, I can set the values of x, y, z of the new origin of the grid, and upon clicking "save", it will shift the grid to the new origin.
By Pointer, I just need to click at any point, and grid will be translated there. The closest intersect under the mouse pointer will decide the origin.
On clicking "shiftCartesianGrid", the user can keep clicking to shift the grid, but to stop it, he needs to right-click (as suggested by @kjvbrt ). Now, the grid will no longer move and stay on its new origin. This explains the unchecking of the "shift cartesian grid" checkbox.

Changing the origin of the grid also changes the origin of everything. So, if I change the origin to say, (100, 100, 100), then if I enable "show 3d points", and click on a point, say "100, 0, 0", then its coordinates will be shown as "0, -100, -100".

Reply:
Okay, but if it's right click to disable, then I don't think
1.  this should be a checkbox
2. you should add some text to the dialog to explain what is going on.
3. it would be nice if the x,y,z numbers in the dialog updated when you click 

I'm also not sure about changing where we report 3d points. Maybe as a compromise you could show both, if the grid origin is no longer 0,0,0?
i.e. Absolute coordinates are (100,0,0) and (0,-100,0) compared to grid

Reply:
I made the suggested changes, could you please review it, and as well, trigger the CI/CD?